rpc/documentation/api.md

```diff
@@ -792,6 +837,148 @@ wallet and republished later if it or a double spend are not mined.
+
+**Expected errors:**
+
+**Stability:** Unstable
+
+___
+
+#### `PurchaseTicket`
+
+The `PurchaseTicket` method is used to purchase tickets. It can use a specified 
+address for voting rights, and can additionally be used in conjunction with a 
+stake pool. An expiration value can be set for the tickets. Expired tickets are 
+pruned from the wallet and funds and then restored to the user. The following 
+fields are considered optional, and a default value given below will be used 
+if they are not specified: ticker_address, pool_address, expiry, tx_fee, 
+ticket_fee.
+
+**Request:** `PurchaseTicketRequest`
+
+- `uint32 account`: The account to use to purchase the tickets.
+
+- `int64 spend_limit`: The maximum amount to pay for a single ticket. If the 
+  current stake difficulty is about this amount, the wallet will return an error.
+
+- `uint32 required_confirmations`: The number of required confirmations for 
+  funds used to purchase a ticket. This field is optional. If left unset, it 
+  will use unconfirmed and confirmed outputs to purchase tickets.
+
+- `string ticket_address`: The address to give voting rights to. This field is 
+  optional. An internal address will be used from the wallet if it is not 
+  specified.
+
+- `uint32 num_tickets`: The number of tickets to purchase. It must be set and 
+  at least 1.
+
+- `string pool_address`: The address of the stake pool used. This field is 
+  optional and pool mode will be disabled if it is not passed.
+
+- `double pool_fees`: The stake pool fees amount. This must be set to a positive 
+  value in the allowed range of 0.01 to 100.00 to be valid. It must be set when 
+  the pool_address is also set.
+
+- `uint32 expiry`: The height at which the tickets expire and can no longer enter 
+  the blockchain. This field is optional. It defaults to 0 (no expiry).
+
+- `int64 tx_fee`: Fees per kB to use for the transaction generating outputs to use 
+  for buying tickets. This field is optional. If this is not passed, the global 
+  value for a transaction fee will be used.
+
+- `int64 ticket_fee`: Fees per kB to use for all purchased tickets. This field is 
+  optional. If not passed, the global value for a ticket fee will be used.
+
+**Response:** `PurchaseTicketResponse`
+
+- `repeated bytes ticket_hashes`: The transaction hashes of the generated tickets.
+
+**Expected errors:**
+
+- `InvalidArgument`: The spent limit was negative.
+
+- `InvalidArgument`: An invalid number of tickets was specified.
+
+- `InvalidArgument`: An invalid ticket address was specified.
+
+- `InvalidArgument`: An invalid pool address was specified.
+
+- `InvalidArgument`: Pool address was specified, but pool fees were not.
+
+- `InvalidArgument`: Pool address was not specified, but pool fees were.
+
+- `InvalidArgument`: And invalid pool fees amount was given, either too large or 
+  small.
+
+- `InvalidArgument`: A negative fees per kB was set.
+
+- `FailedPrecondition`: The wallet is currently open.
```

Comment:
uh, not sure how that got in there. i'll fix it and see if i can figure out what i meant to put.